### ISSUE TITLE
Failed: H2SpecIntegrationSpec

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#1837]
https://jenkins.akka.io:8498/job/akka-http-nightly-against-akka-master/438/testReport/akka.http.h2spec/H2SpecIntegrationSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

```
Error Message
org.scalatest.exceptions.TestFailedException: The future returned an exception of type: java.lang.IllegalStateException, with message: IO Listener actor terminated unexpectedly for remote endpoint [127.0.0.1:34861].
Stacktrace
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: The future returned an exception of type: java.lang.IllegalStateException, with message: IO Listener actor terminated unexpectedly for remote endpoint [127.0.0.1:34861].
	at org.scalatest.concurrent.Futures$FutureConcept$class.tryTryAgain$1(Futures.scala:523)
	at org.scalatest.concurrent.Futures$FutureConcept$class.futureValueImpl(Futures.scala:550)
	at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
	at org.scalatest.concurrent.Futures$FutureConcept$class.futureValue(Futures.scala:476)
	at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
	at akka.http.h2spec.H2SpecIntegrationSpec.afterTermination(H2SpecIntegrationSpec.scala:148)
	at akka.testkit.AkkaSpec.afterAll(AkkaSpec.scala:86)
	at org.scalatest.BeforeAndAfterAll$$anonfun$1.apply$mcV$sp(BeforeAndAfterAll.scala:225)
	at org.scalatest.Status$$anonfun$withAfterEffect$1.apply(Status.scala:379)
	at org.scalatest.Status$$anonfun$withAfterEffect$1.apply(Status.scala:375)
	at org.scalatest.CompositeStatus.whenCompleted(Status.scala:886)
	at org.scalatest.Status$class.withAfterEffect(Status.scala:375)
	at org.scalatest.CompositeStatus.withAfterEffect(Status.scala:782)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:223)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:58)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:480)
	at sbt.ForkMain$Run$2.call(ForkMain.java:300)
	at sbt.ForkMain$Run$2.call(ForkMain.java:290)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: sbt.ForkMain$ForkError: java.lang.IllegalStateException: IO Listener actor terminated unexpectedly for remote endpoint [127.0.0.1:34861]
	at akka.stream.impl.io.ConnectionSourceStage$$anon$2.akka$stream$impl$io$ConnectionSourceStage$$anon$$receive(TcpStages.scala:99)
	at akka.stream.impl.io.ConnectionSourceStage$$anon$2$$anonfun$preStart$1.apply(TcpStages.scala:61)
	at akka.stream.impl.io.ConnectionSourceStage$$anon$2$$anonfun$preStart$1.apply(TcpStages.scala:61)
	at akka.stream.stage.GraphStageLogic$StageActor.internalReceive(GraphStage.scala:230)
	at akka.stream.stage.GraphStageLogic$StageActor$$anonfun$1.apply(GraphStage.scala:198)
	at akka.stream.stage.GraphStageLogic$StageActor$$anonfun$1.apply(GraphStage.scala:198)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:447)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:465)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:560)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:742)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:757)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:517)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:667)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:590)
	at akka.actor.ActorCell.invoke(ActorCell.scala:559)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly-against-akka-master-on-2.12/422/testReport/

**Comment by raboof** [comment]
https://jenkins.akka.io:8498/job/akka-http-nightly-against-akka-master-on-2.12/432//consoleFull

**Comment by jrudolph** [comment]
Fixed by #1839.
